<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1040</width>
    <height>580</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>890</width>
    <height>580</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="styleSheet">
    <string notr="true">background-image: url(:/pic1.jpg)</string>
   </property>
   <widget class="QCustomPlot" name="function_plot" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>561</width>
      <height>471</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>585</x>
      <y>30</y>
      <width>401</width>
      <height>411</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: white;
background-image: url(:/pic2.png)</string>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Triangular</enum>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideLeft</enum>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Standard</string>
     </attribute>
     <widget class="QPushButton" name="StandardPushButtom">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>240</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="StandardLaber">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Enter a normal function</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="StandardTextEdit">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>331</width>
        <height>121</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="XstartLaber">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>180</y>
        <width>47</width>
        <height>12</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:1 rgba(0, 0, 0, 0));</string>
      </property>
      <property name="text">
       <string>X軸起始</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="XstartlineEdit">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>180</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
     </widget>
     <widget class="QLabel" name="XendLaber">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>210</y>
        <width>44</width>
        <height>12</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:1 rgba(0, 0, 0, 0));</string>
      </property>
      <property name="text">
       <string>X軸結束</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="XendlineEdit">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>210</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Polar</string>
     </attribute>
     <widget class="QPushButton" name="PolarPushButtom">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>240</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="PolarLaber">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Enter a polar function</string>
      </property>
     </widget>
     <widget class="QLabel" name="RadiusLaber">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>55</y>
        <width>47</width>
        <height>12</height>
       </rect>
      </property>
      <property name="text">
       <string>Radius</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="PolarlineEdit">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>50</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="maxtheta">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>90</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>90</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>upper limit</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>110</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>( in the unit of π )</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Parameter</string>
     </attribute>
     <widget class="QPushButton" name="ParameterPushButtom">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>240</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="PolarLaber_2">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>20</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Enter 2 parameter functions</string>
      </property>
     </widget>
     <widget class="QLabel" name="Xlabel">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>65</y>
        <width>47</width>
        <height>12</height>
       </rect>
      </property>
      <property name="text">
       <string>X = </string>
      </property>
     </widget>
     <widget class="QLabel" name="Ylabel">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>95</y>
        <width>47</width>
        <height>12</height>
       </rect>
      </property>
      <property name="text">
       <string>Y = </string>
      </property>
     </widget>
     <widget class="QLineEdit" name="ParameterXlineEdit">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>60</y>
        <width>311</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="ParameterYlineEdit">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>90</y>
        <width>311</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="ParameterStartlineEdit">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>130</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="ParameterEndlineEdit">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>160</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="ParameterStartlabel">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>130</y>
        <width>47</width>
        <height>12</height>
       </rect>
      </property>
      <property name="text">
       <string>參數起始</string>
      </property>
     </widget>
     <widget class="QLabel" name="ParameterEndlabel">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>160</y>
        <width>47</width>
        <height>12</height>
       </rect>
      </property>
      <property name="text">
       <string>參數終止</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>data point</string>
     </attribute>
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>181</width>
        <height>311</height>
       </rect>
      </property>
      <row>
       <property name="text">
        <string>1</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>2</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>3</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>4</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>5</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>6</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>7</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>8</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>9</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>10</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>11</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>12</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>13</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>14</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>15</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>16</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>17</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>18</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>19</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>20</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>21</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>22</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>23</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>24</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>25</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>26</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>27</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>28</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>29</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>30</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>31</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>32</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>33</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>34</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>35</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>36</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>37</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>38</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>39</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>40</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>41</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>42</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>43</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>44</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>45</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>46</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>47</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>48</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>49</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>50</string>
       </property>
      </row>
      <column>
       <property name="text">
        <string>X</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Y</string>
       </property>
      </column>
      <item row="46" column="0">
       <property name="text">
        <string/>
       </property>
      </item>
      <item row="48" column="1">
       <property name="text">
        <string/>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>20</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>11</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="text">
       <string>fitting</string>
      </property>
     </widget>
     <widget class="QPushButton" name="DrawPoints">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>360</y>
        <width>181</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>12</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>Draw Points</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>60</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>10</pointsize>
        <italic>true</italic>
       </font>
      </property>
      <property name="text">
       <string>basis</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="FittingBasis">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>80</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="BasisOk">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>220</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>100</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>10</pointsize>
        <italic>true</italic>
       </font>
      </property>
      <property name="text">
       <string>Left endpoint</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="fitting_leftx">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>130</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>160</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>10</pointsize>
        <italic>true</italic>
       </font>
      </property>
      <property name="text">
       <string>Right endpoint</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="fitting_rightx">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>190</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>260</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 87 8pt &quot;Arial Black&quot;;</string>
      </property>
      <property name="text">
       <string>the least squared residuals</string>
      </property>
      <property name="textFormat">
       <enum>Qt::PlainText</enum>
      </property>
      <property name="wordWrap">
       <bool>false</bool>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <property name="indent">
       <number>-1</number>
      </property>
      <property name="buddy">
       <cstring></cstring>
      </property>
     </widget>
     <widget class="QLabel" name="theleastsquared">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>280</y>
        <width>161</width>
        <height>31</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 16pt &quot;Agency FB&quot;;</string>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="FittingHelpButton">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>20</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>help</string>
      </property>
     </widget>
     <widget class="QPushButton" name="AddItem">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>181</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>AddWidget Items</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="saveButtom">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>450</y>
      <width>121</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-image: url(:/pic2.png)</string>
    </property>
    <property name="text">
     <string>Save As</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ShowEquationButton">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>450</y>
      <width>141</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-image: url(:/pic2.png)</string>
    </property>
    <property name="text">
     <string>Show Eqaution</string>
    </property>
   </widget>
   <widget class="QPushButton" name="CursorButton">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>450</y>
      <width>161</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-image: url(:/pic2.png)</string>
    </property>
    <property name="text">
     <string>Show/Hide Cursor</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1040</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
